### ISSUE TITLE
How to reference env variables

### ISSUE THREAD

**Issue by cobrowserAlex** [TheNewNormal/kube-cluster-osx#13]
Sorry for my ignorance but I am trying to us an ENV variable set by kubernetes in a replication container, but all the kubenetes ENV variables come up empty:

```
bash-3.2$ kubectl exec curlrc-mgu06 -- printenv | grep API
API_PORT=tcp://10.100.63.201:8080
API_PORT_8080_TCP_PROTO=tcp
API_PORT_8080_TCP_ADDR=10.100.63.201
API_PORT_8080_TCP_PORT=8080
API_PORT_8080_TCP=tcp://10.100.63.201:8080
API_SERVICE_HOST=10.100.63.201
API_SERVICE_PORT=8080
bash-3.2$ kubectl exec curlrc-mgu06 -- echo $API_SERVICE_HOST

bash-3.2$
```

whereas `echo $HOME` works as expected.

What is the correct way to reference these?

**Comment by rimusz** [comment]
@cobrowserAlex I'm not sure what are you asking there

**Comment by cobrowserAlex** [comment]
@rimusz i want to use the ENV variable API_SERVICE_HOST inside a docker image but the value is always empty. 

So for instance `sed -i "s/api_host/${API_SERVICE_HOST}/g;" /etc/nginx/conf.d/location-api.conf` or any other variant i tried doesn't work since API_SERVICE_HOST is empty.

**Comment by rimusz** [comment]
@cobrowserAlex I do not see the variable API_SERVICE_HOST even on GKE cluster inside the docker container. Try to use KUBERNETES_SERVICE_HOST

**Comment by cobrowserAlex** [comment]
@rimusz Thanks for taking the time to respond, I must be annoyingly ignorant.

I was reading this [documentation](http://kubernetes.io/v1.1/docs/user-guide/connecting-applications.html#environment-variables) and trying something similar, like using   `NGINXSVC_SERVICE_HOST` in a shell script. 

I expect the ENV variable `API_SERVICE_HOST` to exist since i have a service named `api`. it seems to exist because I see the ENV variable in **_any**_ container when I run `printenv` inside the container, like in my op. 

```
API_PORT=tcp://10.100.63.201:8080
API_PORT_8080_TCP_PROTO=tcp
API_PORT_8080_TCP_ADDR=10.100.63.201
API_PORT_8080_TCP_PORT=8080
API_PORT_8080_TCP=tcp://10.100.63.201:8080
API_SERVICE_HOST=10.100.63.201
API_SERVICE_PORT=8080
```

The problem is that when I reference the exposed hostname ENV variable in a shell script, it comes up empty.

**Comment by rimusz** [comment]
But the doc you referred I do not see any `API_PORT` referenced there,  only KUBERNETES_SERVICE_HOST is:

```
$ k exec nginx-s40au -- printenv | grep SERVICE
KUBERNETES_SERVICE_HOST=10.59.240.1
KUBERNETES_SERVICE_PORT=443
```

**Comment by cobrowserAlex** [comment]
@rimusz ok I am not clear in what I am unable to achieve. Let me try again with a simple example.

I have a replication controller setting an ENV variable like this:

```
apiVersion: v1
kind: ReplicationController
metadata:
  name: php-fpm-api
spec:
  replicas: 1
  template:
    metadata:
      labels:
        app: php-fpm-api
    spec:
      containers:
      - name: php-fpm
        image: php:5-cli
        ports:
        - containerPort: 9000
        env:
          - name: MONGO_MAIN_DB
            value: cobrowser_v3
```

when this pod is running, I try this code to reference my MONGO_MAIN_DB env variable:

```
kubectl exec php-fpm-api-zmbb5 -- printenv | grep MONGO_MAIN_DB
MONGO_MAIN_DB=cobrowser_v3
```

This is expected, however referencing it inside a php script doesn't work:

```
kubectl exec php-fpm-api-zmbb5 -- php -r 'var_dump($_ENV["MONGO_MAIN_DB"]);'
PHP Notice:  Undefined index: MONGO_MAIN_DB in Command line code on line 1
NULL
```

**Comment by cobrowserAlex** [comment]
sorry hit the wrong button.

**Comment by rimusz** [comment]
sorry, that has noting to do with the Kubernetes cluster setup.

this is how the php scripts handles env variables.
I'm not the php expert, sorry

**Comment by cobrowserAlex** [comment]
@rimusz ok, thanks anyway.

by the way strange that referencing $HOSTNAME does work:
```
kubectl exec php-fpm-api-zmbb5 -- sh -c 'echo $HOSTNAME'
php-fpm-api-zmbb5
```

**Comment by cobrowserAlex** [comment]
@rimusz thanks!

As usual trying to figure out a way to explain the problem led me to the solution as well as your answer:

```
kubectl exec php-fpm-api-zmbb5 -- php -r 'echo getenv("MONGO_MAIN_DB");'
```

will yield the correct result.

**Comment by rimusz** [comment]
cool, you got it sorted

**Comment by cobrowserAlex** [comment]
For anyone else finding this, I paste the shell variant to get the environment variables. The key is to open a new shell.

```
sh -c 'sed -i "s/domain_placeholder/${DOMAIN_NAME}/g;" /var/www/config/config.js'
```
